comp/core/workloadmeta/collectors/internal/remote/sbomcollector/sbom_collector.go

```diff
@@ -0,0 +1,180 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build trivy
+
+// Package sbomcollector implements the remote SBOM collector for
+// Workloadmeta.
+package sbomcollector
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/CycloneDX/cyclonedx-go"
+	"go.uber.org/fx"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/grpclog"
+
+	"github.com/DataDog/datadog-agent/comp/core/workloadmeta/collectors/internal/remote"
+	workloadmeta "github.com/DataDog/datadog-agent/comp/core/workloadmeta/def"
+	"github.com/DataDog/datadog-agent/pkg/config/model"
+	pkgconfigsetup "github.com/DataDog/datadog-agent/pkg/config/setup"
+	sbompb "github.com/DataDog/datadog-agent/pkg/proto/pbgo/sbom"
+	"github.com/DataDog/datadog-agent/pkg/util/flavor"
+	grpcutil "github.com/DataDog/datadog-agent/pkg/util/grpc"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+const (
+	collectorID       = "sbom-collector"
+	cacheValidityNoRT = 2 * time.Second
+)
+
+type client struct {
+	cl              sbompb.SBOMCollectorClient
+	parentCollector *streamHandler
+}
+
+func (c *client) StreamEntities(ctx context.Context) (remote.Stream, error) {
+	log.Debug("starting a new stream")
+	streamcl, err := c.cl.GetSBOMStream(
+		ctx,
+		&sbompb.SBOMStreamParams{},
+	)
+	if err != nil {
+		return nil, err
+	}
+	return &stream{cl: streamcl}, nil
+}
+
+type stream struct {
+	cl sbompb.SBOMCollector_GetSBOMStreamClient
+}
+
+func (s *stream) Recv() (interface{}, error) {
+	log.Trace("calling stream recv")
+	return s.cl.Recv()
+}
+
+type streamHandler struct {
+	model.Reader
+}
+
+// workloadmetaEventFromSBOMEventSet converts the given SBOM message into a workloadmeta event
+func workloadmetaEventFromSBOMEventSet(event *sbompb.SBOMMessage) (workloadmeta.Event, error) {
+	if event == nil {
+		return workloadmeta.Event{}, nil
+	}
+
+	var bom cyclonedx.BOM
+	decoder := cyclonedx.NewBOMDecoder(bytes.NewReader(event.Data), cyclonedx.BOMFileFormatJSON)
+	if err := decoder.Decode(&bom); err != nil {
+		return workloadmeta.Event{}, fmt.Errorf("failed to unmarshal SBOM: %w", err)
+	}
+
+	if event.Kind != string(workloadmeta.KindContainer) {
+		return workloadmeta.Event{}, fmt.Errorf("expected KindContainer, got %s", event.Kind)
+	}
+
+	if event.ID == "" {
+		return workloadmeta.Event{}, fmt.Errorf("expected container ID, got empty")
```

Comment:
```suggestion
		return workloadmeta.Event{}, fmt.Errorf("expected event ID, got empty")
```

Maybe?

Reply:
In this case, the event ID is really the container ID so I suggest to keep it as it as it would be easier to troubleshoot if that happens, does it ok to you ?

Reply:
Absolutely 😄